bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -753,6 +753,44 @@ def _getZODBAnalysesFromReferenceAnalyses(self, objid, criteria):
 
         return analyses
 
+    def analysisHasCalcution(self, objid, analysis):
+        """ returns True or False"""
+
+        analyses = self._getZODBAnalyses(objid)
+        analyses_with_calculation = filter(lambda c: c.getCalculation(),analyses)
+        if analyses_with_calculation:
+            return True
```

Comment:
I think a `for` loop would be better in this case, because we can return True if just one analysis has a calculation attached, e.g.:

```python
for analysis in analyses:
    if analysis.getCalculation():
        return True
return False
```

Reply:
This method is IMO not needed and can be removed